pkg/auth/auth.go

```diff
@@ -244,7 +245,26 @@ func getUserAndPass(opts *LoginOptions, password, userFromAuthFile string) (user
 	}
 	if password == "" {
 		fmt.Fprint(opts.Stdout, "Password: ")
-		pass, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		// Store and restore the terminal status on interruptions to
+		// avoid that the terminal remains in the password state
+		// This is necessary as a workaround for https://github.com/golang/go/issues/31180
+		stdin := int(os.Stdin.Fd())
+		oldState, err := terminal.GetState(stdin)
+		if err != nil {
+			oldState = nil
+		}
+		sigch := make(chan os.Signal, 1)
+		signal.Notify(sigch, os.Interrupt)
+		go func() {
+			for range sigch {
+				if oldState != nil {
+					terminal.Restore(stdin, oldState)
+				}
+				os.Exit(1)
```

Comment:
I don't think we want to call exit so far down in the stack. At least in podman we have handlers that depend on proper function, see `libpod/shutdown`.

Also I have no idea where this code is used besides podman, I think it might be better to return an error so that the caller can decide what to do with it.

Reply:
I'm not sure how to handle this - In the podman repository I couldn't find any signal handlers for `os.Interrupt` except for [machine.go](https://github.com/containers/podman/blob/main/pkg/machine/qemu/machine.go#L491), which also uses `os.Exit(1)`.

I can certainly modify it such that `getUserAndPass` returns a `fmt.Errorf("interrupted")` but I'm not sure if this is the desired behaviour. What do you think?

Reply:
I concur that we should return an error here.